pkg/ebpf/events_derived.go

```diff
@@ -116,3 +122,68 @@ func deriveContainerRemoved(t *Tracee) deriveFn {
 		return trace.Event{}, false, nil
 	}
 }
+
+func deriveDetectHookedSyscall(t *Tracee) deriveFn {
+	return func(event trace.Event) (trace.Event, bool, error) {
+		syscallsAdresses, err := getEventArgUlongArrVal(&event, "syscalls_addresses")
+		if err != nil {
+			return trace.Event{}, false, fmt.Errorf("error parsing syscalls_numbers arg: %v", err)
+		}
+		hookedSyscallData := t.ParseHookedAddresses(syscallsAdresses)
+		de := event
+		de.EventID = int(DetectHookedSyscallsEventID)
+		de.EventName = "hooked_syscalls"
+		de.ReturnValue = 0
+		de.Args = []trace.Argument{
+			{ArgMeta: trace.ArgMeta{Name: "hooked_syscalls", Type: "hookedSyscallData[]"}, Value: hookedSyscallData},
+		}
+		de.ArgsNum = 1
+		return de, true, nil
+	}
+}
+
+func (t *Tracee) ParseHookedAddresses(addresses []uint64) []bufferdecoder.HookedSyscallData {
```

Comment:
I think the word _parse_ does not fit here well.
_Parse_ means that you take raw data and put it in the appropriate format.
I think the word _analyze_ will match better the case, though you might prefer something else.